### PR TITLE
feat(Security): Improvements suggested by OSSF Scorecard

### PR DESCRIPTION
**Pull request by Chocolate-lightning**
Whats new or changed?

- OSSF Best practice has been  self certified and badge added
- OSSF Scoreboard testing has been run across the project
  - Update: Branch protection rules have been tightened up
    - Dismissal of stale PR approvals
    - Prevent self-assigned reviews on created code
    - Actions required to pass before merge
  - New: Enabled CodeQL code scanning in the repo
  - New: Enabled CodeQL code quality GitHub feature to provide suggestions and reviews on PRs'
  - New: Added CODEOWNERS file
  - Update: GitHub Action files have had their permissions scoped with least access required
  - New: Scorecard GHA to test the project against the OSSF Scoreboard standards regularly and publish the badge
- Update: Linters will now exclude the CHANGELOG.md as it is generated by Release Please for us
- New: Added a test directory that can contain utilities and any overarching tests we may want to add in the future
- New: Fuzz testing enabled within the project via fast-check
- Updated: Update existing unit tests and add edge case testing

### CODE REVIEW

File: .github/workflows/scorecard.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
+name: Scorecard supply-chain security
```
Comment (andrewgos) [inline .github/workflows/scorecard.yml]: nitpicky warning:

- lets name it with capital so its consistent with other workflows?  `Scorecard Supply-chain Security`
- also file name to be clearer? ie. `scorecard-security.yml`

File: CODEOWNERS [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# CODEOWNERS
+
+# All files in the repository
+* @andrewgos @Chocolate-lightning
```
Comment (andrewgos) [inline CODEOWNERS]: Do you have permission to modify github teams/group within the organisation? so we can point it to a group instead of individuals. i.e.
`@moodlehq/design-system`
`@moodlehq/design-system-devs`
`@moodlehq/design-system-designers`

**Review by andrewgos — DISMISSED** [review]
looks good! just few minor items I want to share.

With the tests, let's roll with the suggestions, i'll collaborate on top of it

File: .github/workflows/scorecard.yml (hunk 1 above)
Comment (Chocolate-lightning) [reply .github/workflows/scorecard.yml]: Makes sense, will make that change now.

File: CODEOWNERS (hunk 2 above)
Comment (Chocolate-lightning) [reply CODEOWNERS]: Great idea, Teams created and references updated.

**Review by andrewgos — APPROVED** [review]
🚢
